src/bus/mod.rs

```diff
@@ -189,6 +185,9 @@ mod test {
             }
         }
 
+        // Detect double registration with the same range.
+        assert_eq!(bus.register(range, device), Err(Error::DeviceOverlap));
```

Comment:
Can we also check the overlap when the address start is not the same?

Reply:
This overlap is checked right below this one in the tests